src/injector/middleware.go

```diff
@@ -0,0 +1,23 @@
+package injector
+
+import (
+	"github.com/inloco/kafka-elasticsearch-injector/src/metrics"
+	"github.com/inloco/kafka-elasticsearch-injector/src/models"
+	"time"
+)
+
+type instrumentingMiddleware struct {
+	metricsPublisher metrics.MetricsPublisher
+	next             Service
+}
+
+func (s instrumentingMiddleware) Insert(records []*models.Record) error {
+	begin := time.Now()
+	err := s.next.Insert(records)
+	s.metricsPublisher.RecordEndpointLatency(time.Since(begin).Seconds())
```

Comment:
I believe that recording milliseconds instead of seconds would be better. Hardly would take seconds for an operations like this be completed.

Reply:
It doesn't matter in the end, as this records `float64`, not `int`